test/instrumentation/modules/http/aborted-requests-enabled.js

```diff
@@ -420,3 +420,8 @@ function resetAgent (cb) {
   agent._instrumentation.currentTransaction = null
   agent._transport = mockClient(1, cb)
 }
+
+function get () {
+  agent._instrumentation.currentTransaction = null
```

Comment:
How is there a `currentTransaction` set in the first place, when these tests are resetting the agent before each test? 🤔

Reply:
It's somehow leaking from the previous test inside the `listen` callback - but only sometimes. I tried to figure out why but couldn't without extensive debugging.

Reply:
And funny enough, it's only an issue on Node.js 8.x 🤷‍♂️ So there must be something in regards to how `server.listen` works there.

Reply:
I wonder if that's a bug in how we're using the `patch-async.js` stuff, because it patches the internal `net._listen2` method. 🤔

Reply:
I don't think it's related to `patch-async.js` as this also happens when using Async Hooks as far as I ca see